File: js/src/maplestory/api/dto/character/characterAndroidEquipmentPresetDto.ts
import { CharacterAndroidEquipmentFaceDto } from '@src/maplestory/api/dto/character/characterAndroidEquipmentFaceDto';
import { CharacterAndroidEquipmentHairDto } from '@src/maplestory/api/dto/character/characterAndroidEquipmentHairDto';
import {
  CharacterAndroidEquipmentPresetDtoBody
} from '@src/maplestory/api/response/character/characterAndroidEquipmentDtoBody';

/**
 * 캐릭터 안드로이드 프리셋 정보
 */
class CharacterAndroidEquipmentPresetDto {

  /**
   * 안드로이드 명
   */
  androidName: string;

  /**
   * 안드로이드 닉네임
   */
  androidNickname: string;

  /**
   * 안드로이드 아이콘
   */
  androidIcon: string;

  /**
   * 안드로이드 아이템 설명
   */
  androidDescription: string;

  /**
   * 안드로이드 성별
   */
  androidGender: string | null;

  /**
   * 안드로이드 등급
   */
  androidGrade: string;

  /**
   * 안드로이드 피부 명
   */
  androidSkinName: string;

  /**
   * 안드로이드 헤어 정보
   */
  androidHair: CharacterAndroidEquipmentHairDto;

  /**
   * 안드로이드 성형 정보
   */
  androidFace: CharacterAndroidEquipmentFaceDto;

  /**
   * 안드로이드 이어센서 클립 적용 여부
   */
  androidEarSensorClipFlag: string;

  /**
   * 비인간형 안드로이드 여부
   */
  androidNonHumanoidFlag: string;

  /**
   * 잡화상점 기능 이용 가능 여부
   */
  androidShopUsableFlag: string;

  constructor(obj: CharacterAndroidEquipmentPresetDtoBody) {
    const {
      android_name,
      android_nickname,
      android_icon,
      android_description,
      android_gender,
      android_grade,
      android_skin_name,
      android_hair,
      android_face,
      android_ear_sensor_clip_flag,
      android_non_humanoid_flag,
      android_shop_usable_flag,
    } = obj;

    this.androidName = android_name;
    this.androidNickname = android_nickname;
    this.androidIcon = android_icon;
    this.androidDescription = android_description;
    this.androidGender = android_gender;
    this.androidGrade = android_grade;
    this.androidSkinName = android_skin_name;
    this.androidHair = new CharacterAndroidEquipmentHairDto(android_hair);
    this.androidFace = new CharacterAndroidEquipmentFaceDto(android_face);
    this.androidEarSensorClipFlag = android_ear_sensor_clip_flag;
    this.androidNonHumanoidFlag = android_non_humanoid_flag;
    this.androidShopUsableFlag = android_shop_usable_flag;
  }
}

export { CharacterAndroidEquipmentPresetDto };

File: java/src/main/java/dev/spiralmoon/maplestory/api/tms/dto/character/CharacterAndroidEquipmentPresetDTO.java
package dev.spiralmoon.maplestory.api.tms.dto.character;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

/**
 * 預設的機器人資訊
 */
@AllArgsConstructor
@NoArgsConstructor
@Data
@ToString
public class CharacterAndroidEquipmentPresetDTO implements dev.spiralmoon.maplestory.api.common.dto.character.CharacterAndroidEquipmentPresetDTO<CharacterAndroidEquipmentHairDTO, CharacterAndroidEquipmentFaceDTO, CharacterAndroidEquipmentSkinDTO> {

    /**
     * 機器人名稱
     */
    @JsonProperty("android_name")
    private String androidName;

    /**
     * 機器人暱稱
     */
    @JsonProperty("android_nickname")
    private String androidNickname;

    /**
     * 機器人圖示
     */
    @JsonProperty("android_icon")
    private String androidIcon;

    /**
     * 機器人道具描述
     */
    @JsonProperty("android_description")
    private String androidDescription;

    /**
     * 機器人性別
     */
    @JsonProperty("android_gender")
    private String androidGender;

    /**
     * 機器人階級
     */
    @JsonProperty("android_grade")
    private String androidGrade;

    /**
     * 機器人髮型資訊
     */
    @JsonProperty("android_hair")
    private CharacterAndroidEquipmentHairDTO androidHair;

    /**
     * 機器人臉型資訊
     */
    @JsonProperty("android_face")
    private CharacterAndroidEquipmentFaceDTO androidFace;

    /**
     * 機器人膚色資訊
     */
    @JsonProperty("android_skin")
    private CharacterAndroidEquipmentSkinDTO androidSkin;

    /**
     * 機器人耳飾感應器
     */
    @JsonProperty("android_ear_sensor_clip_flag")
    private String androidEarSensorClipFlag;

    /**
     * 非人型機器人
     */
    @JsonProperty("android_non_humanoid_flag")
    private String androidNonHumanoidFlag;

    /**
     * 是否具備雜貨店功能
     */
    @JsonProperty("android_shop_usable_flag")
    private String androidShopUsableFlag;
}
